src/inet/UDPEndPointImplLwIP.cpp

```diff
@@ -66,27 +63,63 @@ static_assert(LWIP_TCPIP_CORE_LOCKING, "CHIP requires config LWIP_TCPIP_CORE_LOC
 namespace chip {
 namespace Inet {
 
-namespace {
-/**
- * @brief
- * RAII locking for LwIP core to simplify management of
- * LOCK_TCPIP_CORE()/UNLOCK_TCPIP_CORE() calls.
- */
-class ScopedLwIPLock
+EndpointQueueFilter * UDPEndPointImplLwIP::sQueueFilter = nullptr;
+
+err_t EndPointStateLwIP::RunOnTCPIPRet(std::function<err_t()> fn)
```

Comment:
This should really live in EndPointStateLwIP.cpp...

Reply:
ideally yes but given how build rules are paramterized, i'd also need to add empty files for other `${chip_system_config_inet}` values: `EndPointStateSockets.cpp`, `EndPointStateNetworkFramework.cpp`, `EndPointStateOpenThread.cpp` - all empty.
my reasoning was that UDP endpoint is mandatory and TCP is optional, so we can get away with placing this small common bit here, rather than have multiple empty files like that.
but if you'd like me to do that, let me know.

Reply:
I'd think you could just add the new file inside `if (chip_system_config_use_lwip) {` in the BUILD.gn.  That's the condition under which `chip_system_config_inet = "LwIP"` happens....

Reply:
doh! again, please excuse my unfamiliarity with the build system, it's new to me. done.

Reply:
No need to apologize! Build systems are always a pain.